File: backend/src/main/java/org/akashihi/mdg/entity/Currency.java
package org.akashihi.mdg.entity;

import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.Hibernate;

import javax.persistence.Entity;
import javax.persistence.Id;

@Getter
@Setter
@ToString
@Entity
public class Currency {
    @Id
    private Long id;
    private String code;
    private String name;
    private Boolean active;

    @Override
    @SuppressFBWarnings(value = "BC_EQUALS_METHOD_SHOULD_WORK_FOR_ALL_OBJECTS", justification = "Checked with Hibernate.getClass()")
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) {
            return false;
        }
        Currency currency = (Currency) o;
        return id != null && id.equals(currency.id);
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}

File: backend/src/main/kotlin/org/akashihi/mdg/entity/Currency.kt
package org.akashihi.mdg.entity

import org.hibernate.Hibernate
import javax.persistence.Entity
import javax.persistence.Id

@Entity
class Currency(
    var code: String,
    var name: String,
    var active: Boolean,
    @Id
    val id: Long? = null
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) {
            return false
        }
        val currency = other as Currency
        return id != null && id == currency.id
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}
